### PR TITLE
Converted VG to  hierarchical

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
## Checklist for adding MMTEB dataset

Reason for dataset addition:
VG Clustering was not hierarchical nor ClusteringFast before #656 

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by x-tabdeveloping** [comment]
I can't run it for E5 as Ucloud is down today and it would take hours on my computer :')

**Comment by x-tabdeveloping** [comment]
Also added stratified subsampling code to `AbsTask` for multilabel problems, as this was missing.

File: mteb/abstasks/AbsTask.py [hunk 1]
```diff
@@ -16,6 +18,32 @@
 logger = logging.getLogger(__name__)
 
 
+def _multilabel_subsampling(
```
Comment (KennethEnevoldsen) [inline mteb/abstasks/AbsTask.py:21]: Ahh great!

File: mteb/tasks/Clustering/nob/VGHierarchicalClustering.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+from mteb.abstasks import TaskMetadata
+from mteb.abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
+
+
+def split_labels(record: dict) -> dict:
+    record["labels"] = record["labels"].split(",")[:2]
+    return record
+
+
+class VGHierarchicalClusteringP2P(AbsTaskClusteringFast):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/nob/VGHierarchicalClustering.py:12]: Should we outdate VG?

File: mteb/abstasks/AbsTask.py [hunk 3]
```diff
@@ -66,7 +94,15 @@ def stratified_subsampling(
         """
         ## Can only do this if the label column is of ClassLabel.
         if not isinstance(dataset_dict[splits[0]].features[label], datasets.ClassLabel):
-            dataset_dict = dataset_dict.class_encode_column(label)
+            try:
```
Comment (KennethEnevoldsen) [inline mteb/abstasks/AbsTask.py:97]: Let us just validate this in existing fast clustering tasks

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good. Let us get the results in from the other models as well - otherwise good

File: mteb/tasks/Clustering/nob/VGHierarchicalClustering.py (hunk 2 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Clustering/nob/VGHierarchicalClustering.py:12]: I think I added superseeded_by to the OG task

**Comment by x-tabdeveloping** [comment]
Since the stratified subsampling doesn't exactly work as expected with multilabel data, I will just go with a random sample of 2048 entries I think.

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good!

**Comment by x-tabdeveloping** [comment]
Tests are failing because of missing datasets unrelated to my PR, what to do?

**Pull request merged**